scripts/code_generation/jsoncan/src/codegen/c_generation/app_can_alerts_module.py

```diff
@@ -108,6 +109,57 @@ def _board_has_alert_func(self, alert_type: CanAlertType, comment: str):
         has_alert.body.add_line("return false;")
         return has_alert
 
+    def _get_board_alert_code(self, alert_type: CanAlertType, comment: str):
+        get_alert = CFunc(
+            GET_BOARD_FAULT_CODE.format(alert_type=alert_type),
+            "void",
+            args=[
+                CVar("board", CTypesConfig.CAN_ALERT_BOARD_ENUM),
+                CVar("*alertArray", "uint8_t"),
+                CVar("*p","uint8_t")
+            ],
+            comment=f"Return whether or not a board has set a {comment}.",
+        )
+        get_alert.body.start_switch("board")
+
+        nodes_with_alerts = [
+            node for node in self._db.nodes if self._db.node_has_alert(node, alert_type)
+        ]
+        for node in nodes_with_alerts:
+            get_alert.body.add_switch_case(
+                ALERT_BOARD_ENUM_NAME.format(node=node.upper())
+            )
+            get_alert.body.start_switch_case()
+
+            for alert in self._db.node_alerts_with_rx_check(
+                node, self._node, alert_type
+            ):
+                if node == self._node:
+                    get_alert.body.start_if(
+                        f"{CFuncsConfig.APP_TX_GET_SIGNAL.format(signal=alert)}()"
+                    )
+                else:
+                    get_alert.body.start_if(
+                        f"{CFuncsConfig.APP_RX_GET_SIGNAL.format(signal=alert)}()"
+                    )
+                get_alert.body.add_line("alertArray= (uint8_t *) realloc(alertArray, sizeof(uint8_t));")
+                get_alert.body.add_line("alertArray[*p-1] = (uint8_t)" + alert + ";")
+                get_alert.body.add_line("*p++;")
+
+                get_alert.body.end_if()
+                get_alert.body.add_line()
```

Comment:
a couple things about this function:
1. `p` is not a great variable name, since its not very clear what it means. usually i think single letter names should be avoided (except for iterators, like `i` or `j`).
2. `realloc` here allocates new data on the heap, which is bad! in embedded programming, its generally bad practice to use the heap at runtime because microcontrollers have very little RAM and its very easy to fill the heap up. its also very easy to write memory bugs that are hard to catch/test for (for example, i don't think you ever deallocate the the memory you've allocated, which means eventually the heap will fill up and this will fail!). also, its usually just unnecessary. that is why you pass in an array here so this function is provided with the memory it needs rather than allocating its own.